sdk/inc/azure/core/az_json.h

```diff
@@ -303,6 +320,34 @@ AZ_NODISCARD az_result az_json_builder_append_bool(az_json_builder* json_builder
 AZ_NODISCARD az_result
 az_json_builder_append_int32_number(az_json_builder* json_builder, int32_t value);
 
+/**
+ * @brief Appends a double number value.
+ *
+ * @param[in] json_builder A pointer to an #az_json_builder instance containing the buffer to append
+ * the number to.
+ * @param[in] value The value to be written as a JSON number.
+ * @param[in] fractional_digits The number of digits of the \p value to write after the decimal
+ * point and truncate the rest.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if the number was appended successfully
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the buffer is too small
+ *         - #AZ_ERROR_NOT_SUPPORTED if the \p value contains an integer component that is too
+ * large and would overflow beyond 2^53 - 1
+ *
+ * @remark Only finite double values are supported. Values such as NAN and INFINITY are not allowed
+ * and would lead to invalid JSON being written.
+ *
+ * @remark Non-significant trailing zeros (after the decimal point) are not written, even if \p
+ * fractional_digits is large enough to allow the zero padding.
+ *
+ * @remark The \p fractional_digits must be between 0 and 15 (inclusive). Any value passed in that
+ * is larger will be clamped down to 15.
+ */
+AZ_NODISCARD az_result az_json_builder_append_double_number(
```

Comment:
Are you renaming json builder to json reader?

Reply:
json builder -> json writer
json parser -> json reader

Yes, in a separate, standalone PR.